### ISSUE TITLE
Videostream-cli webcam to another videostream-cli  not working

### ISSUE THREAD

**Issue by AnasMK** [aiortc/aiortc#257]
Thank you for this amazing package.

I am trying to use this package for streaming on raspberry pi to Ubuntu machine receiving frames and handle it with opencv like in server example. 
Tried those on raspberry pi:
1. Server example works fine 
2. Webcam works fine
3. Videostream-cli example, streaming an mp4 video works fine and timing is correct 
But running videostream-cli with webcam instead of video doesn't work, I receive first one or couple of frames correctly then the stream get stopped hang or receiving  curropted frames. 
* I had tried different type of cameras
* and test on raspberry pi 3 and 4
* tried different versions of ffmpeg even downloading source and compile it
Non of them was able to stream successfully.

Why is this happening since both sever and webcam both works fine, means PYAV and FFMPEG installed correctly. 

This how I am running videostream-cli:
on raspberry pi: 
`python cli.py offer`
and this how I modified the code to make offer stream from webcam:
```
# create media source
    if args.play_from:
        player = MediaPlayer(args.play_from)
    else:
        if args.role == "offer":
            options = {"framerate": "30", "video_size": "640x480"}
            player = MediaPlayer("/dev/video0", format="v4l2", options=options)
        else:
            player = None
````
on Ubuntu machine I run 
`python cli.py answer --record-to video.mp4`

**Comment by jlaine** [comment]
Does the unmodified example work?

**Comment by AnasMK** [comment]
Sorry for late reply,
Yes, the unmodified example works fine.
On Rasspberry pi I run 
`python cli.py offer - - play-from video.mp4`
On Ubuntu machine `python cli.py answer - -record-to record.mp4`
The recorded video is fine and the timing is correct

**Comment by jlaine** [comment]
I just tried and was unable to reproduce the issue on my laptop: the answerer receives video and the video is written to file as expected.

**Comment by AnasMK** [comment]
Did you try it with raspberry pi as the offer and your laptop is the answerer?
Did use raspberry pi webcam or mp4 video?

**Comment by jlaine** [comment]
No I do not have a raspberry pi. Bear in mind this is my free time I'm giving at no charge. Don't expect me to go out and buy gear to reproduce issues.

**Comment by AnasMK** [comment]
Thanks alot for your time and efforts.
I've investigate a lot in the issue before I submit it here, even I have tried different FFMPEG versions and compile some of them from source.
I've submitted the issue here for hope that you or someone come across or face  the same problem using raspberry pi. 

I appreciate your valuable time for developing, maintaining and open sourcing such package. Thanks alot again.
